JavaScript/JavaScript_Limitations.md

```diff
@@ -0,0 +1,85 @@
+# The Not So Good Parts of JavaScript
+
+JavaScript is quite an amazing programming language, despite what some in the development community have to say about it.  It runs in web browsers, which is great as it's executed on the client side, as opposed to the server, making it faster for the end user.  It makes web pages feel 'alive' with its interactivity.  However there are also problems, the fact it runs on the web browser makes it a security issue, as this can be used to exploit the users system.  This article is not a definitive guide of all the limitations of JavaScript, it does cover some of the more common points as a developer you should be aware of.
+
+#### Multithreading
+
+Firstly JavaScript does not support multithreading.  Multithreading is a way to manage multiple events, known as threads simultaneously, so that different stages of a program run at the same time without interfering with each other.  Most modern programming languages like Java and C# support multithreading, so intensive code maybe executed.  JavaScript runs in a single execution thread, this is a design feature which you cannot change.  However multithreading may be simulated, so code which would freeze up the browser can be run.  Using asynchronous timers which can run repetitive code at staggered times so the browser will have interpreter time to process separate iterations.  HTML5 Web workers are multithreaded and allow a single javascript thread to run, thus giving the appearance that it is multithreaded.
+
```

Comment:
Is it worth noting that Brendan Eich said 'over my dead body' and 'I'd rather eat glass' than add multithreading to JS?